src/components/RightSidebar/TableOfContents.tsx

```diff
@@ -12,28 +12,28 @@ interface Props {
 	isMobile?: boolean;
 }
 
-const TableOfContents: FunctionalComponent<Props> = ({ toc = [], labels, isMobile }) => {
+const TableOfContents = ({ toc = [], labels, isMobile }: Props) => {
 	const [currentHeading, setCurrentHeading] = useState({
 		slug: toc[0].slug,
 		text: toc[0].text,
 	});
 	const [open, setOpen] = useState(!isMobile);
 	const onThisPageID = 'on-this-page-heading';
 
-	const Container = ({ children }) => {
+	const Container = ({ children }: { children: JSX.Element }) => {
 		return isMobile ? (
-			<details {...{ open }} onToggle={(e) => setOpen(e.target.open)} class="toc-mobile-container">
+			<details {...{ open }} onToggle={(e) => setOpen(e.target.open)} className="toc-mobile-container">
 				{children}
 			</details>
 		) : (
 			children
 		);
 	};
 
-	const HeadingContainer = ({ children }) => {
+	const HeadingContainer = ({ children }: { children: JSX.Element }) => {
```

Comment:
Was supposed to use `ComponentChildren` here per https://preactjs.com/guide/v10/typescript#function-components, but will run into another issue:

<img width="767" alt="CleanShot 2023-01-19 at 14 40 41@2x" src="https://user-images.githubusercontent.com/1091472/213373214-89146541-dfec-4f2a-bc07-cf4711e4936f.png">

So again, this is a hack to work around types discrepancies between React and Preact.